<launch>
  <arg name="publish_tf" default="true"/>
  <node name="orb_slam2_mono" pkg="orb_slam2_ros"
      type="orb_slam2_ros_mono" args="
          $(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt
          $(find orb_slam2_ros)/orb_slam2/config/Astra.yaml"
      output="screen">
       <remap from="/camera/image_raw" to="/camera/rgb/image_raw" />

       <param name="publish_pointcloud" type="bool" value="true" />
       <param name="publish_pose" type="bool" value="true" />
       <param name="publish_tf" type="bool" value="$(arg publish_tf)" />
       <param name="localize_only" type="bool" value="false" />
       <param name="reset_map" type="bool" value="false" />
       <param name="pointcloud_frame_id" type="string" value="map" />
       <param name="camera_frame_id" type="string" value="camera_rgb_frame" />
       <param name="min_num_kf_in_map" type="int" value="5" />

       <param name="publish_map_to_odom" type="bool" value="true"/>                                                                                                                                        
       <param name="robot_base_link_frame_id" type="string" value="base_link"/>      
  </node>
</launch>
